src/Util.vala

```diff
@@ -192,6 +192,38 @@ namespace Tasks.Util {
     }
 
 
+    /**
+     * Returns the value of X-EVOLUTION-GTASKS-POSITION property if set,
+     * otherwise return "00000000000000000000"
+     */
+    public string get_gtasks_position_property_value (ECal.Component ecalcomponent) {
+        unowned ICal.Component? icalcomponent = ecalcomponent.get_icalcomponent ();
+        if (icalcomponent != null) {
+            var gtasks_position = ECal.util_component_dup_x_property (icalcomponent, "X-EVOLUTION-GTASKS-POSITION");
+
+            if (gtasks_position != null) {
+                return gtasks_position;
+            }
+        }
+
+        // returns the default value for a task created without a position
+        return "00000000000000000000";
+    }
+
+    public bool update_gtasks_position (ECal.Component ecalcomponent, string position) {
+        var icomp = ecalcomponent.get_icalcomponent ();
+        if (icomp != null) {
+            var icomp_position = ECal.util_component_dup_x_property (icomp, "X-EVOLUTION-GTASKS-POSITION");
+
+            if (icomp_position != position) {
+                ECal.util_component_set_x_property (icomp, "X-EVOLUTION-GTASKS-POSITION", position);
```

Comment:
Nice catch! Seems like we can simplify some of the Task's code using this neat utility function.